Modules/TPC/src/IDCs.cxx

```diff
@@ -0,0 +1,170 @@
+    if ((keyVec.size() > 0) && (keyVec.size() == valueVec.size())) {
+      for (const auto& key : keyVec) {
+        mStoreMaps.back().insert(std::pair<std::string, std::string>(key, valueVec.at(vecIter)));
+        vecIter++;
+      }
+    }
+    if (keyVec.size() != valueVec.size()) {
+      throw std::runtime_error("Number of keys and values for storeMetaData are not matching");
+    }
+    keyVec.clear();
+    valueVec.clear();
+  }
+
+  for (const auto& entry : config.get_child("qc.postprocessing." + name + ".histogramRanges")) {
+    for (const auto& type : entry.second) {
+      for (const auto& value : type.second) {
+        mRanges[type.first].emplace_back(std::stof(value.second.data()));
+      }
+    }
+  }
+
+  for (const auto& entry : config.get_child("qc.postprocessing." + name + ".timestamps")) {
+    for (const auto& type : entry.second) {
+      mTimestamps[type.first] = std::stol(type.second.data());
+    }
+  }
+
+  mHost = config.get<std::string>("qc.config.conditionDB.url");
+}
+
+void IDCs::initialize(Trigger, framework::ServiceRegistry&)
+{
+  mCdbApi.init(mHost);
+
+  mIDCZeroRadialProf = std::make_unique<TCanvas>("c_sides_IDC0_radialProfile");
+  mIDCZeroStacksA = std::make_unique<TCanvas>("c_GEMStacks_IDC0_1D_ASide");
+  mIDCZeroStacksC = std::make_unique<TCanvas>("c_GEMStacks_IDC0_1D_CSide");
+
+  mIDCDeltaStacksA = std::make_unique<TCanvas>("c_GEMStacks_IDCDelta_1D_ASide");
+  mIDCDeltaStacksC = std::make_unique<TCanvas>("c_GEMStacks_IDCDelta_1D_CSide");
+
+  mIDCOneSides1D = std::make_unique<TCanvas>("c_sides_IDC1_1D");
+
+  mFourierCoeffsA = std::make_unique<TCanvas>("c_FourierCoefficients_1D_ASide");
+  mFourierCoeffsC = std::make_unique<TCanvas>("c_FourierCoefficients_1D_CSide");
+
+  getObjectsManager()->startPublishing(mIDCZeroRadialProf.get());
+  getObjectsManager()->startPublishing(mIDCZeroStacksA.get());
+  getObjectsManager()->startPublishing(mIDCZeroStacksC.get());
+
+  getObjectsManager()->startPublishing(mIDCDeltaStacksA.get());
+  getObjectsManager()->startPublishing(mIDCDeltaStacksC.get());
+
+  getObjectsManager()->startPublishing(mIDCOneSides1D.get());
+
+  getObjectsManager()->startPublishing(mFourierCoeffsA.get());
+  getObjectsManager()->startPublishing(mFourierCoeffsC.get());
+}
+
+void IDCs::update(Trigger, framework::ServiceRegistry&)
+{
+  mCCDBHelper.setIDCZero(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCZero>("TPC/Calib/IDC/IDC0", std::map<std::string, std::string>{}, mTimestamps["IDCZero"]));
+  mCCDBHelper.setIDCDelta(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCDelta<float>>("TPC/Calib/IDC/IDCDELTA", std::map<std::string, std::string>{}, mTimestamps["IDCDelta"]));
+  mCCDBHelper.setIDCOne(mCdbApi.retrieveFromTFileAny<o2::tpc::IDCOne>("TPC/Calib/IDC/IDC1", std::map<std::string, std::string>{}, mTimestamps["IDCOne"]));
+  mCCDBHelper.setFourierCoeffs(mCdbApi.retrieveFromTFileAny<o2::tpc::FourierCoeff>("TPC/Calib/IDC/FOURIER", std::map<std::string, std::string>{}, mTimestamps["FourierCoeffs"]));
```

Comment:
I assume these values are only needed for the drawing. Then you can do:

```suggestion
  auto idcZero = mCdbApi.retrieveFromTFileAny<o2::tpc::IDCZero>("TPC/Calib/IDC/IDC0", std::map<std::string, std::string>{}, mTimestamps["IDCZero"]);
  auto idcDelta = mCdbApi.retrieveFromTFileAny<o2::tpc::IDCDelta<float>>("TPC/Calib/IDC/IDCDELTA", std::map<std::string, std::string>{}, mTimestamps["IDCDelta"]);
  auto idcOne = mCdbApi.retrieveFromTFileAny<o2::tpc::IDCOne>("TPC/Calib/IDC/IDC1", std::map<std::string, std::string>{}, mTimestamps["IDCOne"]);
  auto idcFFT = mCdbApi.retrieveFromTFileAny<o2::tpc::FourierCoeff>("TPC/Calib/IDC/FOURIER", std::map<std::string, std::string>{}, mTimestamps["FourierCoeffs"]);

  mCCDBHelper.setIDCZero(idcZero);
  mCCDBHelper.setIDCDelta(idcDelta);
  mCCDBHelper.setIDCOne(idcOne);
  mCCDBHelper.setFourierCoeffs(idcFFT);
```

Reply:
Yes, only for drawing. This should work.